### ISSUE TITLE
Training Details

### ISSUE THREAD

**Issue by sgadgil6** [yudaleng/COPD-Early-Prediction#1]
Can you please provide the training details used to train DeepSpiro and the CatBoostClassifier? How many epochs, learning rate, optimizer was used? Were all the modules trained sequentially or end-to-end?

**Comment by yudaleng** [comment]
In DeepSpiro training, we used the following parameters:
> batch_size: 256
> epochs: 31
> learning rate: 0.001
> optimizer: Adam

In the training of CatBoostClassifier, we used the default parameters for training. Throughout the program, we trained the modules sequentially.

**Comment by sgadgil6** [comment]
Thank you for the quick response and the parameter details! 

I was still a bit confused on the sequential training - from what I understand, the SpiroEncoder outputs an initial COPD assessment which is then fused with demographic information and passed through the SpiroExplainer to get the final COPD risk. Are the SpiroEncoder and SpiroExplainer trained separately using the same COPD labels by standard cross-entropy loss (when training the SpiroExplainer, the SpiroEncoder is frozen)? Or only the final COPD risk output of the SpiroExplainer is used to train both?

Also, how are the ground truth labels used to train the future risk disease predictor obtained?

**Comment by yudaleng** [comment]
> Thank you for the quick response and the parameter details! 
> 
> 
> 
> I was still a bit confused on the sequential training - from what I understand, the SpiroEncoder outputs an initial COPD assessment which is then fused with demographic information and passed through the SpiroExplainer to get the final COPD risk. Are the SpiroEncoder and SpiroExplainer trained separately using the same COPD labels by standard cross-entropy loss (when training the SpiroExplainer, the SpiroEncoder is frozen)? Or only the final COPD risk output of the SpiroExplainer is used to train both?
> 
> 
> 
> Also, how are the ground truth labels used to train the future risk disease predictor obtained? 

We first trained the SpiroEncoder to output the initial copd assessment using the COPD reality tag. We then combined the initial COPD assessment with demographic information to train SpiroExplainer to obtain model interpretability diagram and COPD risk values. Finally, we trained the SpiroPredictor with COPD risk values combined with key acceleration information.
For the real label of future risk disease predictor, please refer to the ukbb Category 1712, which is the information about the time of diagnosis of the patient, through the information about the time of diagnosis of the patient and the time of lung function test, we can roughly determine whether the patient will have COPD in the future.
I hope my answer can help you, wish you have a good day!

**Comment by sgadgil6** [comment]
Thank you for the prompt response, that makes things clearer! One final question I had - what do you mean by the COPD reality tag, is it the COPD label?

**Comment by yudaleng** [comment]
> Thank you for the prompt response, that makes things clearer! One final question I had - what do you mean by the COPD reality tag, is it the COPD label? 

Yes, I'm sorry for the misunderstanding. Thank you for correcting it
